api/base/filters.py

```diff
@@ -106,22 +130,47 @@ def query_params_to_odm_query(self, query_params):
             query = None
         return query
 
-    # Used so that that queries by _id will work
-    def convert_key(self, key):
-        key = key.strip()
-        if self.serializer_class._declared_fields[key].source:
-            return self.serializer_class._declared_fields[key].source
-        return key
 
-    # Used to convert string values from query params to Python booleans when necessary
-    def convert_value(self, value):
-        value = value.strip()
-        if value in self.TRUTHY:
-            return True
-        elif value in self.FALSY:
-            return False
-        # Convert me to current user's pk
-        elif value == 'me' and not self.request.user.is_anonymous():
-            return self.request.user.pk
+class ListFilterMixin(FilterMixin):
+    """View mixin that adds a get_queryset_from_request method which uses query params
+    of the form `filter[field_name]=value` to filter a list of objects.
+
+    Subclasses must define `get_default_queryset()`.
+
+    Serializers that want to restrict which fields are used for filtering need to have a variable called
+    filterable_fields which is a frozenset of strings representing the field names as they appear in the serialization.
+    """
+
+    def get_default_queryset(self):
+        raise NotImplementedError('Must define get_default_queryset')
+
+    def get_queryset_from_request(self):
+        default_queryset = self.get_default_queryset()
+        if self.request.QUERY_PARAMS:
+            param_queryset = self.param_queryset(self.request.QUERY_PARAMS, default_queryset)
+            return param_queryset
         else:
-            return value
+            return default_queryset
+
+    def param_queryset(self, query_params, default_queryset):
+        """filters default queryset based on query parameters"""
+        fields_dict = query_params_to_fields(query_params)
+        return_val = default_queryset
+        if fields_dict:
+            for field_name, value in fields_dict.items():
+                if self.is_filterable_field(key=field_name):
+                    if isinstance(self.serializer_class._declared_fields[field_name], ser.SerializerMethodField):
+                        return_val = self.get_serializer_field_value(field_name, value, default_queryset)
+                    elif isinstance(self.serializer_class._declared_fields[field_name], ser.BooleanField):
+                        return_val = [item for item in default_queryset if item.get('key', None) == value]
```

Comment:
It's not a database query at this point; it's a list that's part of the detail node that's already loaded. This is why she had to split it out into a new mixin rather than just using the existing one.